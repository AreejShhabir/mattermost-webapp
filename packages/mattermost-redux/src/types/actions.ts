// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {ThunkAction} from 'redux-thunk';
import {AnyAction} from 'redux';
import {BatchAction} from 'redux-batched-actions';

import {GlobalState} from '@mattermost/types/store';
import {ServerError} from '@mattermost/types/errors';

export type GetStateFunc = () => GlobalState;
export type GenericAction = AnyAction;
export type Thunk = (b: DispatchFunc, a: GetStateFunc) => Promise<ActionResult> | ActionResult;

export type Action = GenericAction | Thunk | BatchAction | ActionFunc;

export type ActionResult<Data = unknown, Error = unknown> = {
    data?: Data;
    error?: Error;
};

export type DispatchFunc = (
    action: Action,
    getState?: GetStateFunc | null
) => Promise<ActionResult>;

export type ActionFunc<Data = any, State = GlobalState, Error = ServerError, ExtraThunkArg = null> = ThunkAction<
Promise<ActionResult<Data, Error>> | ActionResult<Data, Error>,
State,
ExtraThunkArg,
AnyAction
>;
