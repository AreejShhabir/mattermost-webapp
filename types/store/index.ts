// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {Channel} from '@mattermost/types/channels';
import {GlobalState as BaseGlobalState} from '@mattermost/types/store';
import {RelationOneToOne} from '@mattermost/types/utilities';

import {NewPostDraft} from './draft';
import {PluginsState} from './plugins';
import {ViewsState} from './views';

export type DraggingState = {
    state?: string;
    type?: string;
    id?: string;
}

export type GlobalState = BaseGlobalState & {
    plugins: PluginsState;
    storage: {
        storage: Record<string, any>;
        initialized: boolean;
    };
    views: ViewsState;
    drafts: {
        byChannel: RelationOneToOne<Channel, NewPostDraft>;
        byThread: RelationOneToOne<Channel, NewPostDraft>;
    };
};
