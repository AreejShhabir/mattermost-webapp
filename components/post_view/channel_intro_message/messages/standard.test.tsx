// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {Channel} from '@mattermost/types/channels';

import StandardIntroMessage from './standard';

describe('components/post_view/ChannelIntroMessages', () => {
    const channel = {
        create_at: 1508265709607,
        creator_id: 'creator_id',
        delete_at: 0,
        display_name: 'test channel',
        header: 'test',
        id: 'channel_id',
        last_post_at: 1508265709635,
        name: 'testing',
        purpose: 'test',
        team_id: 'team-id',
        type: 'O',
        update_at: 1508265709607,
    } as Channel;

    // type PluginComponent
    const boardComponent = {
        id: 'board',
        pluginId: 'board',
    };

    const baseProps = {
        currentUserId: 'test-user-id',
        channel,
        fullWidth: true,
        locale: 'en',
        channelProfiles: [],
        enableUserCreation: false,
        teamIsGroupConstrained: false,
        creatorName: 'creatorName',
        stats: {},
        usersLimit: 10,
        actions: {
            getTotalUsersStats: jest.fn().mockResolvedValue([]),
        },
    };

    describe('test Open Channel', () => {
        test('should match snapshot, without boards', () => {
            const wrapper = shallow(
                <StandardIntroMessage{...baseProps}/>,
            );
            expect(wrapper).toMatchSnapshot();
        });

        test('should match snapshot, with boards', () => {
            const wrapper = shallow(
                <StandardIntroMessage
                    {...baseProps}
                    boardComponent={boardComponent}
                />,
            );
            expect(wrapper).toMatchSnapshot();
        });
    });
});
