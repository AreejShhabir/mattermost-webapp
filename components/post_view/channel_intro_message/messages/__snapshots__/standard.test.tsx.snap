// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/post_view/ChannelIntroMessages test Open Channel should match snapshot, with boards 1`] = `
<Fragment>
  <h2
    className="channel-intro__title"
  >
    <MemoizedFormattedMessage
      defaultMessage="Beginning of {name}"
      id="intro_messages.beginning"
      values={
        Object {
          "name": "test channel",
        }
      }
    />
  </h2>
  <p
    className="channel-intro__content"
  >
    <span>
      <MemoizedFormattedMessage
        defaultMessage="This is the start of the {name} channel, created by {creator} on {date}."
        id="intro_messages.creator"
        values={
          Object {
            "creator": "creatorName",
            "date": <FormattedDate
              day="2-digit"
              month="long"
              value={1508265709607}
              year="numeric"
            />,
            "name": "test channel",
          }
        }
      />
    </span>
    <MemoizedFormattedMessage
      defaultMessage=" Any member can join and read this channel."
      id="intro_messages.anyMember"
    />
    <span>
      <MemoizedFormattedMessage
        defaultMessage=" This channel's purpose is: {purpose}"
        id="intro_messages.purpose"
        values={
          Object {
            "purpose": "test",
          }
        }
      />
    </span>
    <br />
  </p>
  <Memo(AddMembersButton)
    channel={
      Object {
        "create_at": 1508265709607,
        "creator_id": "creator_id",
        "delete_at": 0,
        "display_name": "test channel",
        "header": "test",
        "id": "channel_id",
        "last_post_at": 1508265709635,
        "name": "testing",
        "purpose": "test",
        "team_id": "team-id",
        "type": "O",
        "update_at": 1508265709607,
      }
    }
    createBoard={
      <Memo(BoardsButton)
        boardComponent={
          Object {
            "id": "board",
            "pluginId": "board",
          }
        }
        channel={
          Object {
            "create_at": 1508265709607,
            "creator_id": "creator_id",
            "delete_at": 0,
            "display_name": "test channel",
            "header": "test",
            "id": "channel_id",
            "last_post_at": 1508265709635,
            "name": "testing",
            "purpose": "test",
            "team_id": "team-id",
            "type": "O",
            "update_at": 1508265709607,
          }
        }
      />
    }
    setHeader={
      <Memo(Connect(ChannelPermissionGate))
        channelId="channel_id"
        permissions={
          Array [
            "manage_public_channel_properties",
          ]
        }
        teamId="team-id"
      >
        <Memo(SetHeaderButton)
          channel={
            Object {
              "create_at": 1508265709607,
              "creator_id": "creator_id",
              "delete_at": 0,
              "display_name": "test channel",
              "header": "test",
              "id": "channel_id",
              "last_post_at": 1508265709635,
              "name": "testing",
              "purpose": "test",
              "team_id": "team-id",
              "type": "O",
              "update_at": 1508265709607,
            }
          }
        />
      </Memo(Connect(ChannelPermissionGate))>
    }
    usersLimit={10}
  />
</Fragment>
`;

exports[`components/post_view/ChannelIntroMessages test Open Channel should match snapshot, without boards 1`] = `
<Fragment>
  <h2
    className="channel-intro__title"
  >
    <MemoizedFormattedMessage
      defaultMessage="Beginning of {name}"
      id="intro_messages.beginning"
      values={
        Object {
          "name": "test channel",
        }
      }
    />
  </h2>
  <p
    className="channel-intro__content"
  >
    <span>
      <MemoizedFormattedMessage
        defaultMessage="This is the start of the {name} channel, created by {creator} on {date}."
        id="intro_messages.creator"
        values={
          Object {
            "creator": "creatorName",
            "date": <FormattedDate
              day="2-digit"
              month="long"
              value={1508265709607}
              year="numeric"
            />,
            "name": "test channel",
          }
        }
      />
    </span>
    <MemoizedFormattedMessage
      defaultMessage=" Any member can join and read this channel."
      id="intro_messages.anyMember"
    />
    <span>
      <MemoizedFormattedMessage
        defaultMessage=" This channel's purpose is: {purpose}"
        id="intro_messages.purpose"
        values={
          Object {
            "purpose": "test",
          }
        }
      />
    </span>
    <br />
  </p>
  <Memo(AddMembersButton)
    channel={
      Object {
        "create_at": 1508265709607,
        "creator_id": "creator_id",
        "delete_at": 0,
        "display_name": "test channel",
        "header": "test",
        "id": "channel_id",
        "last_post_at": 1508265709635,
        "name": "testing",
        "purpose": "test",
        "team_id": "team-id",
        "type": "O",
        "update_at": 1508265709607,
      }
    }
    createBoard={
      <Memo(BoardsButton)
        channel={
          Object {
            "create_at": 1508265709607,
            "creator_id": "creator_id",
            "delete_at": 0,
            "display_name": "test channel",
            "header": "test",
            "id": "channel_id",
            "last_post_at": 1508265709635,
            "name": "testing",
            "purpose": "test",
            "team_id": "team-id",
            "type": "O",
            "update_at": 1508265709607,
          }
        }
      />
    }
    setHeader={
      <Memo(Connect(ChannelPermissionGate))
        channelId="channel_id"
        permissions={
          Array [
            "manage_public_channel_properties",
          ]
        }
        teamId="team-id"
      >
        <Memo(SetHeaderButton)
          channel={
            Object {
              "create_at": 1508265709607,
              "creator_id": "creator_id",
              "delete_at": 0,
              "display_name": "test channel",
              "header": "test",
              "id": "channel_id",
              "last_post_at": 1508265709635,
              "name": "testing",
              "purpose": "test",
              "team_id": "team-id",
              "type": "O",
              "update_at": 1508265709607,
            }
          }
        />
      </Memo(Connect(ChannelPermissionGate))>
    }
    usersLimit={10}
  />
</Fragment>
`;
